src/hotspot/share/gc/g1/g1CodeRootSet.cpp

```diff
@@ -189,7 +189,8 @@ class G1CodeRootSetHashTable : public CHeapObj<mtGC> {
     // A table with the new size should be at most filled by this factor. Otherwise
     // we would grow again quickly.
     const float WantedLoadFactor = 0.5;
-    size_t min_expected_size = checked_cast<size_t>(ceil(current_size / WantedLoadFactor));
+    assert((current_size / WantedLoadFactor) <= SIZE_MAX, "table overflow");
+    size_t min_expected_size = current_size / WantedLoadFactor;
```

Comment:
Is it UB or compiler-specific behavior if the double --> size_t overflows?  Because gcc returns SIZE_MAX.